ext/flexmailer/src/ClickTracker/TextClickTracker.php

```diff
@@ -37,7 +37,7 @@ public static function replaceTextUrls($text, $replace) {
     };
     // Find any HTTP(S) URLs in the text.
     // return preg_replace_callback('/\b(?:(?:https?):\/\/|www\.|ftp\.)[-A-Z0-9+&@#\/%=~_|$?!:,.]*[A-Z0-9+&@#\/%=~_|$]/i', $callback, $tex
-    return preg_replace_callback('/\b(?:(?:https?):\/\/)[-A-Z0-9+&@#\/%=~_|$?!:,.{}\[\];]*[A-Z0-9+&@#\/%=~_|${}\[\];]/i',
+    return preg_replace_callback('/\b(?:(?:https?):\/\/)[\w+&@#\/%=~_|$?!:,.{}\[\];]*[\w+&@#\/%=~_|${}\[\];]/iu',
```

Comment:
ok - so that's something I should try to remember - `\w` is unicode inclusive so *better* than the a-z

One thing - the old versio is only upper I think? why is that?

Reply:
so \w respects the \u unicode global modifier but -A-Z0-9 doesn't

Reply:
@seamuslee001 OK - I think I'd better let @totten check the regex too

Reply:
> the old versio[n] is only upper I think? why is that?

Old version has `/i` at the end, so it was case-insensitive. `a-z` and `A-Z` mean the same thing in that context.

Keeping the `/i` when using `\w` is maybe a little superfluous, but I think the `/i` still helps for matching `http://` and `HTTP://` (which is non-conventional... but humans are non-conventional...).